server/src/test/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileControllerTest.java

```diff
@@ -0,0 +1,274 @@
+
+        when(memberProfileRepository.findByPdlId(testUUuid)).thenReturn(result);
+
+        final HttpResponse<?> response = client.toBlocking().exchange(HttpRequest.GET(String.format("/?pdlId=%s", testUUuid)));
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertEquals(2, response.getContentLength());
+    }
+
+    @Test
+    public void testGet() {
+
+        Date testDate = new Date(System.currentTimeMillis());
+        String testUser = "testName";
+        String testRole = "testRole";
+        UUID testPdlId = UUID.randomUUID();
+        Map<String, Object> fakeBody = new HashMap<String, Object>() {{
+            put("name", testUser);
+            put("role", testRole);
+            put("pdlId", testPdlId);
+            put("location", "testLocation");
+            put("workEmail", "testEmail");
+            put("insperityId", "testInsperityId");
+            put("startDate", testDate);
+            put("bioText", "testBio");
+        }};
+
+        // Save
+        client.toBlocking().exchange(HttpRequest.POST("", fakeBody));
+
+        // test Find All
+        HttpRequest requestFindAll = HttpRequest.GET(String.format(""));
+        List<MemberProfile> responseFindAll = client.toBlocking().retrieve(requestFindAll, Argument.of(List.class, MemberProfile.class));  
+        assertEquals(1, responseFindAll.size());
+        assertEquals(testUser, responseFindAll.get(0).getName());
+        assertEquals(testRole, responseFindAll.get(0).getRole());
+
+        // test Find By Name
+        HttpRequest requestFindByName = HttpRequest.GET(String.format("/?name=%s", testUser));
+        List<MemberProfile> responseFindByName = client.toBlocking().retrieve(requestFindByName, Argument.of(List.class, MemberProfile.class));  
+        assertEquals(1, responseFindByName.size());
+        assertEquals(testUser, responseFindByName.get(0).getName());
+        assertEquals(testRole, responseFindByName.get(0).getRole());
+
+        // test Find By Role
+        HttpRequest requestFindByRole = HttpRequest.GET(String.format("/?role=%s", testRole));
+        List<MemberProfile> responseFindByRole = client.toBlocking().retrieve(requestFindByRole, Argument.of(List.class, MemberProfile.class));  
+        assertEquals(1, responseFindByRole.size());
+        assertEquals(testUser, responseFindByRole.get(0).getName());
+        assertEquals(testRole, responseFindByRole.get(0).getRole());
+
+        // test Find By PdlId
+        HttpRequest requestFindByPdlId = HttpRequest.GET(String.format("/?pdlId=%s", testPdlId));
+        List<MemberProfile> responseFindByPdlId = client.toBlocking().retrieve(requestFindByPdlId, Argument.of(List.class, MemberProfile.class));  
+        assertEquals(1, responseFindByPdlId.size());
+        assertEquals(testUser, responseFindByPdlId.get(0).getName());
+        assertEquals(testRole, responseFindByPdlId.get(0).getRole());
+        assertEquals(testPdlId, responseFindByPdlId.get(0).getPdlId());
```

Comment:
I would tend to split these into separate tests.  If one fails at some point, it can be beneficial to know if only that case has regressed or multiple cases have regressed. Bundling then together will hide that, as the first failure will kill the test.